### PR TITLE
Allow setting the texture pixel format and type

### LINKED ISSUE #298: StageXL doesn't upload TexturePacker atlases as RGBA 4444
Texture Packer outputs the following info in its json:

```json
"meta": {
	"app": "http://www.codeandweb.com/texturepacker",
	"version": "1.0",
	"image": "assets_main@2x.png",
	"format": "RGBA4444",
	"scale": "1",
}

```

Even though this info is present, StageXL doesn't upload as RGBA 4444, thus missing 2x reduction in GPU memory!

### PR DESCRIPTION
**Pull request by mnordine**
Closes #298.

### CODE REVIEW

**Comment by mnordine** [comment]
Not sure how this resulted in a conflict, but it seems pretty obvious to fix.

**Comment by mnordine** [comment]
Note this automatically reads and sets the pixel formats on sprite sheets produced by Texture Packer. It does not do so for other formats, but they should be fairly easy to set after this PR.

File: lib/src/resources/texture_atlas_format_json.dart [hunk 1]
```diff
@@ -15,6 +15,9 @@ class _TextureAtlasFormatJson extends TextureAtlasFormat {
     final image = meta['image'] as String;
     final renderTextureQuad = await loader.getRenderTextureQuad(image);
 
+    //  Set texture info based on meta format
+    _setTextureFormat(renderTextureQuad.renderTexture, meta['format'] as String);
```
Comment (kevmoo) [inline lib/src/resources/texture_atlas_format_json.dart:19]: Yes! The value `format` may be null – at least for pop-pop-win!

Comment (mnordine) [reply lib/src/resources/texture_atlas_format_json.dart:19]: Understood, will fix, most likely this evening...

Comment (mnordine) [reply lib/src/resources/texture_atlas_format_json.dart:19]: PR here: https://github.com/bp74/StageXL/pull/355

**Pull request merged**